src/types.ts

```diff
@@ -189,7 +189,7 @@ interface Array<T> {
    */
   findAndReplace(
     predicate: (value: T, index: number, obj: T[]) => boolean,
-    replaceVal: T,
+    replaceVal: T | ((x: T) => Partial<T>),
```

Comment:
You should probably create a test for using the function instead of T as replace value as well.

Reply:
It wasn't meant to be part of this PR. I'll remove it.